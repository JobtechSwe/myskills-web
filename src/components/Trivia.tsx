import React from 'react'
import { H1, Paragraph, Italic } from 'components/Typography'
import Grid from 'components/Grid'
import questionMarkIllustration from 'assets/illustrations/question_mark.svg'

interface TriviaProps {
  title: string
  info: string
  source: string
}

const Trivia: React.FC<TriviaProps> = ({ title, info, source }) => {
  return (
    <Grid justifyContent="center" justifyItems="center" gridGap="medium">
      <img alt="Frågetecken" src={questionMarkIllustration} />
      <H1 mb={0} color="white">
        {title}
      </H1>
      <Paragraph color="white">{info}</Paragraph>
      <Italic color="white">Källa: {source}</Italic>
    </Grid>
  )
}

export default Trivia
