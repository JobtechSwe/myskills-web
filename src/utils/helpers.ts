import Cookies from 'js-cookie'
import { LocalStateProps, ContactInformationProps } from '../graphql/client'
import {
  EducationInput,
  Occupation,
  OntologyRelationResponse,
} from '../generated/myskills'

export const getCookie = (name: string) => Cookies.get(name)

export const setCookie = (
  name: string,
  value: string,
  options: object = {}
) => {
  Cookies.set(name, value, options)
}

export const removeCookie = (name: string) => {
  Cookies.remove(name)
}

export const redirect = (route: string) => (location.href = route)

interface StorageEntryProps {
  type: string
}

interface Educations extends StorageEntryProps {
  type: 'educations'
  data: EducationInput[]
}

interface OccupationStorage extends StorageEntryProps {
  type: 'occupation'
  data: Occupation
}

interface Skills extends StorageEntryProps {
  type: 'skills'
  data: OntologyRelationResponse[]
}
interface Traits extends StorageEntryProps {
  type: 'traits'
  data: string[]
}
interface WhoAmI extends StorageEntryProps {
  type: 'whoAmI'
  data: string
}

interface ContactInformation extends StorageEntryProps {
  type: 'contact'
  data: ContactInformationProps
}

export type StorageEntry =
  | ContactInformation
  | Educations
  | OccupationStorage
  | Skills
  | Traits
  | WhoAmI

export const storageHelper = {
  load: (initialState: LocalStateProps) =>
    Object.entries(initialState).reduce(
      (acc, [prop, val]) => ({
        ...acc,
        [prop]: JSON.parse(localStorage.getItem(prop) as string) || val,
      }),
      {} as LocalStateProps
    ),
  set: (payload: StorageEntry) =>
    localStorage.setItem(payload.type, JSON.stringify(payload.data)),
}
