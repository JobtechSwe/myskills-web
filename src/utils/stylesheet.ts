const colors = {
  lassekongo: 'dodgerblue',
  primary: 'pink',
}

const sizes = ['0', '1rem', '2rem', '4rem']

const typography = {
  default: 'Arial',
}

const space = {
  none: sizes[0],
  small: sizes[1],
  medium: sizes[2],
  big: sizes[3],
}

export default { colors, space, typography }
