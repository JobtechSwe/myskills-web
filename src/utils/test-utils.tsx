import * as React from 'react'
import ApolloClient from 'apollo-client'
import { ApolloProvider } from 'react-apollo-hooks'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { MockLink, MockedProvider } from 'react-apollo/test-utils'
import { ThemeProvider } from 'emotion-theming'
import { render as rtlRender } from 'react-testing-library'
import { theme } from 'theme'

export const render = (children: React.ReactNode, mocks?: any[]) => {
  const client = new ApolloClient({
    cache: new InMemoryCache({ addTypename: true }),
    link: new MockLink(mocks || []),
  })

  return rtlRender(
    <MockedProvider addTypename={true} mocks={mocks}>
      <ApolloProvider client={client}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </ApolloProvider>
    </MockedProvider>
  )
}
