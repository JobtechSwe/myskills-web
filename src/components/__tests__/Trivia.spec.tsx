import * as React from 'react'
import Trivia from '../Trivia'
import { render } from 'utils/test-utils'

describe('components/Trivia', () => {
  it('renders Trivia', async () => {
    const { container } = render(
      <Trivia
        info="Yrkesprognosen för lärare ser god ut inom de närmaste åren."
        source="acme.se"
        title="Visste du att?"
      />
    )

    expect(container).toMatchSnapshot()
  })
})
