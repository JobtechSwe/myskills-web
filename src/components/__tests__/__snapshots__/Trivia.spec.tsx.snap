// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Trivia renders Trivia 1`] = `
<div>
  <div
    class="css-71ynm3"
  >
    <img
      alt="Frågetecken"
      src="question_mark.svg"
    />
    <h1
      class="css-ei8wm3"
      color="white"
      font-family="default"
      font-size="large"
      font-weight="600"
    >
      Visste du att?
    </h1>
    <p
      class="css-1d428vd"
      color="white"
      font-family="default"
      font-size="small"
    >
      Yrkesprognosen för lärare ser god ut inom de närmaste åren.
    </p>
    <p
      class="css-1496y5v"
      color="white"
      font-family="default"
      font-size="medium"
      font-weight="300"
    >
      Källa: 
      acme.se
    </p>
  </div>
</div>
`;
